packages/contractkit/src/wrappers/DowntimeSlasher.ts

```diff
@@ -0,0 +1,83 @@
+import { Address } from '../base'
+import { DowntimeSlasher } from '../generated/types/DowntimeSlasher'
+import {
+  BaseWrapper,
+  CeloTransactionObject,
+  proxyCall,
+  toTransactionObject,
+  valueToInt,
+} from './BaseWrapper'
+
+/**
+ * Contract handling slashing for Validator downtime
+ */
+export class DowntimeSlasherWrapper extends BaseWrapper<DowntimeSlasher> {
+  getEpochNumberOfBlock = proxyCall(
+    this.contract.methods.getEpochNumberOfBlock,
+    undefined,
+    valueToInt
+  )
+
+  slashableDowntime = proxyCall(this.contract.methods.slashableDowntime, undefined, valueToInt)
+
+  isDown = proxyCall(this.contract.methods.isDown)
+
+  async slashValidator(
+    validatorAddress: Address,
+    startBlock: number
+  ): Promise<CeloTransactionObject<void>> {
+    const validators = await this.kit.contracts.getValidators()
+    const validator = await validators.getValidator(validatorAddress)
+    return this.slash(
+      startBlock,
+      validators.findSignerIndex(await validators.getSignersForBlock(startBlock), validator.signer)
+    )
+  }
+
+  async slash(endBlock: number, endSignerIndex: number): Promise<CeloTransactionObject<void>> {
+    const validators = await this.kit.contracts.getValidators()
+    const signer = await this.contract.methods
+      .validatorSignerAddressFromSet(endSignerIndex, endBlock)
+      .call()
+    const endEpoch = await this.getEpochNumberOfBlock(endBlock)
+    // Reverses DowntimeSlasher.getEndBlock()
+    const startBlock = endBlock - (await this.slashableDowntime()) + 1
+    const startEpoch = await this.getEpochNumberOfBlock(startBlock)
+    const startSignerIndex =
+      startEpoch === endEpoch
+        ? endSignerIndex
+        : validators.findSignerIndex(await validators.getSignersForBlock(startBlock), signer)
+
+    const validator = await validators.getValidatorFromSigner(signer)
```

Comment:
Maybe we can ask `LockedGold` wrapper to compute all the parameters required to slash a given penalty,user... and thus don't duplicate the logic here and on the other slashing wrapper.

LockedGold seems to be the right choice, since it's the contract that actually does the slashing.

something like `computeParametersForSlashing()`

Reply:
Added computeParametersForSlashing() to LockedGold